sklearn/metrics/pairwise.py

```diff
@@ -544,11 +543,7 @@ def cosine_distances(X, Y=None):
     S *= -1
     S += 1
     np.clip(S, 0, 2, out=S)
-    if X is Y or Y is None:
```

Comment:
Hmm... Since we are preserving existing behaviour, we shouldn't be raising a warning in Euclidean even if the distances are bigger than tolerance

Reply:
Sorry, are you telling me that I roll back behavior of the Euclidean or Cosine distances?

Reply:
Where we are already setting the diagonal to zero without warning we should continue to do so